### PR TITLE
Return Promises from Chat API

### PR DESCRIPTION
**Pull request by thebearjew**
This PR updates the chat bot API to return promises for async actions. It additionally adds some moderate Flow typing where there was none and also resolves a few flow type issues.

Branch is rebased on top of `nathan/one-shot`.

### CODE REVIEW

**Comment by nathunsmitty** [comment]
lgtm! Just a couple of questions

File: demos/check_for_unreads.js [hunk 1]
```diff
@@ -3,25 +3,15 @@ const {Bot} = require('../index.js')
...
-          console.log(
-            'You have ' + unreadCount + ' unread conversations out of ' + res.conversations.length + ' total.'
-          )
-        }
-      })
-    }
-  }
-)
+  })
+  .then(() => {
```
[the remark this answers is not in the document]
Comment (thebearjew) [reply demos/check_for_unreads.js]: @nathunsmitty Yep! Agree, just need to set the minimum Node version (7.10.1 I think)

File: lib/bot.js [hunk 2]
```diff
@@ -45,49 +64,40 @@ class Bot {
...
   // --------------------------------------------------------------------------
 
-  chatList(options: any, cb: CbAny): void {
-    if (options) {
-      throw new Error('options (arg1) for chatList not yet supported. Please pass null.')
-    }
-    this._safelyRunApiCommand({method: 'list', options: {}}, (err, res) => {
-      return cb(err, res)
-    })
+  chatList(options: ChatListOptions): Promise<any> {
```
[the remark this answers is not in the document]
Comment (thebearjew) [reply lib/bot.js]: @nathunsmitty I agree. I'm going to have to go through and type all the input options defined here: https://github.com/keybase/client/blob/master/go/client/chat_api_handler.go

Also need to attempt to type the return values which are a bit harder unless we want `string`

**Comment by nathunsmitty** [comment]
Ok everything looks like it's working for me! @thebearjew wanna test and lemme know what you think?

**Comment by thebearjew** [comment]
@nathunsmitty this part looks good to me, fire away

**Comment by thebearjew** [comment]
@nathunsmitty Actually have to update the README accordingly

**Pull request merged**